lib/Interpreter/Transaction.cpp

```diff
@@ -62,12 +62,22 @@ namespace cling {
   }
 
   NamedDecl* Transaction::containsNamedDecl(llvm::StringRef name) const {
-    for (auto I = decls_begin(), E = decls_end(); I < E; ++I)
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
-        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI))
+        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
           if (name.equals(ND->getNameAsString()))
             return ND;
+        }
+        else if (LinkageSpecDecl* LSD = dyn_cast<LinkageSpecDecl>(DI)) {
```

Comment:
What is the current DeclContext for a Transaction?

I'm happy to submit a patch that does two separate passes, as you are right this can lead to a substantial penalty in it's current form (inclusion of any C headers for example).